### PR TITLE
1108 nice to haves

### PR DESCRIPTION
**Pull request by raduiordache**
Fixes #<issue>

## Checklist:
- [x] My code follows the pykechain style
    - [x] I added type hinting to all functions
    - [x] I added documentation for all public functions
    - [x] I locally used `tox` to check for dists and docs errors
    - [x] My code passes the `pep8` and `flake8` linting checks and no warnings
    - [x] I removed unused imports, using `Code > Optimize Imports` in PyCharm
    - [x] I used [`black`](https://github.com/psf/black) to format the new code
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] I committed fresh test cassettes
    - [x] I have proper test coverage (don't decline the coverage of the test)
- [x] I asked another teammate to review the code
- [x] I update the Changelog.md with the appropriate changes.

### CODE REVIEW

File: pykechain/models/form.py [hunk 1]
```diff
@@ -478,3 +477,52 @@ def apply_transition(self, transition: Transition):
...
+
+        url = self._client._build_url("forms_has_part",
+                                      form_id=self.id,
+                                      part_id=part_id)
+        response = self._client._request(
+            "GET", url=url
+        )
+        if response.status_code != requests.codes.ok:
+            raise APIError(
+                "Could not process whether `Form` {} has part {}".format(self.id, part_id),
```
Comment (jberends) [inline pykechain/models/form.py:502]: please use the f-strings

```suggestion
                f"Could not process whether `Form` {self.id} has part {part_id}",
```

File: pykechain/models/form.py [hunk 2]
```diff
@@ -478,3 +477,52 @@ def apply_transition(self, transition: Transition):
...
+    def has_part(self, part: Part) -> bool:
+        """Return boolean if given Part is part of the Form tree.
+
+        Based on the Parts category, either the model or the instance tree is checked.
+
+        :param part: a Part object
+        :raises IllegalArgumentError: in case not a `Part` object is used when calling the method
+        :raises APIError: in case an Error occurs when checking whether the `Form` contains the
+        `Part`
+
```
Comment (jberends) [inline pykechain/models/form.py:490]: drop line

```suggestion
```

File: pykechain/models/form.py [hunk 3]
```diff
@@ -478,3 +477,52 @@ def apply_transition(self, transition: Transition):
...
+        """
+        part_id = check_base(part, Part)
+
+        url = self._client._build_url("forms_has_part",
```
Comment (jberends) [inline pykechain/models/form.py:494]: did you use black here?

**Review by jberends — CHANGES_REQUESTED** [review]
small changes

File: pykechain/models/form.py (hunk 3 above)
Comment (raduiordache) [reply pykechain/models/form.py]: I did now

File: CHANGELOG.rst [hunk 4]
```diff
@@ -9,6 +9,8 @@ FORMS Feature
 * :+1: Added 'Forms', 'Form models', 'Contexts' and 'Workflows' sidebar buttons in the `SideBarManager' and added ability to specify 'minimum_access_level' and 'alignment' per 'SideBarButton' (#1103)
 * :+1: Added the `Forms` class and tests (#1100)
 * :+1: Added multiple `Forms` methods such as `link_contexts`, `unlink_contexts`, `clone_cross_scope`, `apply_transition`, `set_status_assignees` (#1114)
+* :+1: Added nice to have `Forms` methods such as `has_part` and `compatible_within_scope` and tests (#1108)
+: :+1: Removed the `target_scope` parameter when cloning inside the same scope
```
Comment (jberends) [inline CHANGELOG.rst:13]:
```suggestion
* :+1: Removed the `target_scope` parameter when cloning inside the same scope
```

**Review by jberends — APPROVED** [review]

**Pull request merged**